### PR TITLE
Cleanup and Refactoring

### PR DESCRIPTION
**Pull request by ishbir**
Separated object header encoding/decoding, added CalcInventoryHash, made Address.CalcDoubleHash private, ran go fmt, golint and go vet.

### CODE REVIEW

File: wire/msgbroadcast_test.go [hunk 1]
```diff
@@ -114,14 +114,6 @@ func TestBroadcastWire(t *testing.T) {
...
 	wireErr := &wire.MessageError{}
 
-	// Ensure calling MsgVersion.Decode with a non *bytes.Buffer returns
-	// error.
-	fr := newFixedReader(0, []byte{})
-	if err := baseMsg.Decode(fr); err == nil {
-		t.Errorf("Did not receive error when calling " +
-			"MsgVersion.Decode with non *bytes.Buffer")
-	}
-
```
Comment (DanielKrawisz) [inline wire/msgbroadcast_test.go]: Why did you remove this test (and a few others like it)?

Comment (ishbir) [inline wire/msgbroadcast_test.go]: It's a redundant test.. the first test in every tests struct already checks for when the number of bytes available is 0. It makes no difference to the test coverage.

Comment (DanielKrawisz) [inline wire/msgbroadcast_test.go]: Ok, I think I understand what happened now. You really cleaned everything up nicely I think!
